pymc_experimental/utils/model_fgraph.py

```diff
@@ -184,8 +184,7 @@ def fgraph_from_model(
 
     # Replace RNG nodes so that seeding does not interfere with old model
     for rng in find_rng_nodes(model_vars):
-        new_rng = rng.clone()
-        new_rng.set_value(rng.get_value(borrow=False))
+        new_rng = shared(rng.get_value(borrow=False))
```

Comment:
I don't understand why the old approach didn't work. You were creating a clone, then setting its value to a new generator. Why would wrapping a new shared around a generator work and the old approach not.

Reply:
A sharedvariable.clone reuses the same container. The only thing that's cloned is the `Variable` object: https://github.com/pymc-devs/pytensor/blob/673c1accf98659ba0457759431cb36eef4659f63/pytensor/compile/sharedvalue.py#L139-L149